samples/ordered_puzzle/components/target_box.js

```diff
@@ -0,0 +1,29 @@
+AFRAME.registerComponent("target-box", {
+  schema: {
+    jsonData: {
+      parse: JSON.parse,
+      stringify: JSON.stringify,
+    },
+  },
+  multiple: true,
+  /**
+   * Initial creation and setting of the mesh.
+   */
+  init: function () {
+    const data = this.data.jsonData;
+    const el = this.el;
+
+    // Create geometry.
+    this.geometry = new THREE.PlaneGeometry(data.width, data.height);
+
+    // Create material.
+    this.material = new THREE.MeshStandardMaterial({ color: data.color });
+
+    // Create mesh.
+    this.mesh = new THREE.Mesh(this.geometry, this.material);
+
+    el.object3D.position.set(data.x, data.y, data.z);
```

Comment:
In accordance with above comments, this will have to change to like `data.xTarget` instead of `data.x` etc